src/wallet/adapters/ConnectAdaptor.ts

```diff
@@ -31,71 +32,148 @@ export class ConnectAdaptor implements AUTHAdapter {
     baseUrl
   }: ConnectAdaptorConfig) {
     this.chainId = chainId
-    this.userName = userName
+    this.passkeyName = passkeyName
     this.API = new API(apiKey, +chainId, baseUrl)
     this.provider = new StaticJsonRpcProvider(
       rpcUrl ? rpcUrl : networks[+this.chainId].RPCUrl
     )
   }
 
   async connect(walletAddress?: string): Promise<void> {
-    if (walletAddress) await this._verifywalletAddress(walletAddress)
-
     this.projectParams = await this.API.getProjectParams()
 
     const isWebAuthnCompatible = await webAuthnService.isWebAuthnCompatible()
 
-    if (!isWebAuthnCompatible) {
-      this.signer = walletAddress
-        ? await burnerWalletService.getSigner(
-            this.API,
-            this.provider,
-            walletAddress
-          )
-        : await burnerWalletService.createSignerAndWallet(this.API)
+    if (walletAddress) {
+      const wallet = await this.getWalletInfos(walletAddress)
+      if (!wallet) throw new Error('Wallet does not exists')
+
+      if (isWebAuthnCompatible) {
+        const { publicKeyId, signerAddress } = await webAuthnService.getSigner({
+          API: this.API,
+          walletAddress
+        })
+        this.signer = new WebAuthnSigner(publicKeyId, signerAddress)
+      } else {
+        this.signer = await burnerWalletService.getSigner({
+          API: this.API,
+          provider: this.provider,
+          walletAddress
+        })
+      }
+
+      this.walletAddress = walletAddress
     } else {
-      try {
-        const { publicKeyId, signerAddress } = walletAddress
-          ? await webAuthnService.getSigner(this.API, walletAddress)
-          : await webAuthnService.createSignerAndWallet(this.API, this.userName)
+      if (isWebAuthnCompatible) {
+        const {
+          publicKeyX,
+          publicKeyY,
+          publicKeyId,
+          signerAddress,
+          deviceData,
+          walletAddress
+        } = await webAuthnService.createSigner({
+          API: this.API,
+          passkeyName: this.passkeyName
+        })
 
         this.signer = new WebAuthnSigner(publicKeyId, signerAddress)
-      } catch {
-        throw new Error(
-          'New Domain detected. You need to add that domain as signer.'
-        )
+        this.walletAddress = walletAddress
+
+        await this.API.initWalletWithWebAuthn({
+          walletAddress,
+          publicKeyId,
+          publicKeyX,
+          publicKeyY,
+          deviceData
+        })
+      } else {
+        const { signer, walletAddress } =
+          await burnerWalletService.createSigner({
+            API: this.API
+          })
+
+        this.signer = signer
+        this.walletAddress = walletAddress
+
+        await this.API.initWallet({
+          ownerAddress: signer.address
+        })
       }
     }
+  }
+
+  async importSafe(
+    walletAddress: string,
+    message: string,
+    signature: string
+  ): Promise<string> {
+    const safeVersion = await safeService.getSafeVersion(
```

Comment:
What happen if you pass a non safe wallet address ? IT throw an error ? Need try catch ?